api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -320,28 +321,37 @@ private void discontinueExistingOrdersIfNecessary(Order order) {
 		    null);
 		boolean isDrugOrderAndHasADrug = DrugOrder.class.isAssignableFrom(getActualType(order))
 		        && ((DrugOrder) order).getDrug() != null;
+		Order orderToBeDiscontinued = null;
 		for (Order activeOrder : orders) {
```

Comment:
In fact, i think you need to get rid of this discontinueExistingOrdersIfNecessary() method since this is no longer necessary after all it is called in one palce